repository/src/lib.rs

```diff
@@ -96,8 +97,33 @@ impl<T: RawRepository> DistributedRepository<T> {
     /// - the `a-#` branches
     /// - the `b-#` branches
     /// if only the branches are not outdated (branched from the last finalized commit).
-    pub async fn clean(&self) -> Result<(), Error> {
-        unimplemented!()
+    pub async fn clean(&mut self) -> Result<(), Error> {
+        let finalized_branch_commit_hash =
+            self.raw.locate_branch(FINALIZED_BRANCH_NAME.into()).await?;
+
+        let branches = self.raw.list_branches().await?;
+
+        for branch in branches {
+            let branch = branch.as_str();
+            if !(branch == WORK_BRANCH_NAME
```

Comment:
why not just `branch.as_str() ==` (to save a line, and to reduce the use of `to_string()` down there)